stdlib/src/builtin/float_literal.mojo

```diff
@@ -275,24 +283,39 @@ struct FloatLiteral(
             return self
 
         alias one = __mlir_attr.`#kgen.int_literal<1> : !kgen.int_literal`
+        alias neg_one = __mlir_attr.`#kgen.int_literal<-1> : !kgen.int_literal`
         alias ten = __mlir_attr.`#kgen.int_literal<10> : !kgen.int_literal`
         var multiplier = one
+        var target: Self = self
         # TODO: Use IntLiteral.__pow__() when it's implemented.
-        for _ in range(ndigits):
+        for _ in range(abs(ndigits)):
             multiplier = __mlir_op.`kgen.int_literal.binop`[
                 oper = __mlir_attr.`#kgen<int_literal.binop_kind mul>`
             ](multiplier, ten)
-        var target: Self = self * Self(multiplier)
-        var truncated: Self = target.__int_literal__()
+        if ndigits > 0:
+            target *= Self(multiplier)
+        elif ndigits < 0:
+            target /= Self(multiplier)
+        else:
+            return self.__round__()
+        var truncated: IntLiteral = target.__int_literal__()
         var result: Self
-        if abs(target) - abs(truncated) <= 0.5:
-            result = truncated
-        elif self > 0:
-            result = truncated + 1
+        var abs_diff = abs(target - truncated)
+        var plus_one = one if self > 0 else neg_one
+        if abs_diff == 0.5:
+            # Round to the nearest even number.
+            if truncated % 2 == 0:
+                result = Self(truncated)
```

Comment:
Same as above: maybe early return?

Reply:
This time would be more difficult, as we compute a `result` value just for a final operation in the last lines:
```
... # computing result
if ndigits >= 0:                                                                                                                                                                                                         
    result /= Self(multiplier)                                                                                                                                                                                           
elif ndigits < 0:                                                                                                                                                                                                        
    result *= Self(multiplier)
return result
```